### PR TITLE
Add QUIC transport to the test Swarm

### PR DESCRIPTION
**Pull request by aarshkshah1992**
We need to add a QUIC transport to the testing swarm so peers in tests can listen on/dial to quic addresses.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/libp2p/go-libp2p-core v0.7.0
 	github.com/libp2p/go-libp2p-loggables v0.1.0
 	github.com/libp2p/go-libp2p-peerstore v0.2.6
+	github.com/libp2p/go-libp2p-quic-transport v0.9.1
```
Comment (marten-seemann) [inline go.mod:13]: Please use v0.9.0 for now, 0.9.1 is kind of broken.

File: go.sum [hunk 2]
```diff
@@ -411,9 +575,17 @@ gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
...
+honnef.co/go/tools v0.0.0-20180728063816-88497007e858/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
+honnef.co/go/tools v0.0.0-20190106161140-3f1c8253044a/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.1-2019.2.3 h1:3JgtbtFHMiCmsznwGVTUWbgGov+pVqnlf1dEJTNAXeM=
 honnef.co/go/tools v0.0.1-2019.2.3/go.mod h1:a3bituU0lyd329TUQxRnasdCoJDkEUEAqEt0JzvZhAg=
+rsc.io/quote/v3 v3.1.0/go.mod h1:yEA65RcK8LyAZtP9Kv3t0HmxON59tX3rD+tICJqUlj0=
+rsc.io/sampler v1.3.0/go.mod h1:T1hPZKmBbMNahiBKFy5HrXp6adAjACjK9JXDnKaTXpA=
+sourcegraph.com/sourcegraph/go-diff v0.5.0/go.mod h1:kuch7UrkMzY0X+p9CRK03kfuPQ2zzQcaEFbx8wA8rck=
+sourcegraph.com/sqs/pbtypes v0.0.0-20180604144634-d3ebe8f20ae4/go.mod h1:ketZ/q3QxT9HOBeFhu6RdvsftgpsbFHBF5Cas6cDKZ0=
```
Comment (marten-seemann) [inline go.sum:591]: That seems like a lot of changes. Maybe run `go mod tidy`?

**Review by marten-seemann — APPROVED** [review]

**Comment by Stebalien** [comment]
LGTM but this does bloat the default libp2p dependency tree a bit (although it doesn't affect any produced binaries). @raulk objections?

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
I believe the go version in CI also needs to be updated.

**Comment by Stebalien** [comment]
Fixed in #235.

**Pull request closed without merging**